### ISSUE TITLE
[BUG] Invalid `transactions_hash` value from `block` stream

### ISSUE THREAD

**Issue by rezacute** [hyperledger/iroha#3668]
### OS and Environment

Ubuntu, Docker Hub

### GIT commit hash

3b463093

### Minimum working example / Steps to reproduce

```bash
./iroha_client_cli events pipeline

```

in another shell window
```bash
./iroha_client_cli blocks 1

```

in another shell window
```bash
./iroha_client_cli asset transfer --from mad_hatter@looking_glass --to white_rabbit@looking_glass --asset-id tea#looking_glass --quantity 5
```

### Actual result

```bash
"3B5467F0C33854C00AF89B04D5992474D58D770F7F364F262BEB593916823AD9"
```

```json
{
  "Pipeline": {
    "entity_kind": "Transaction",
    "status": "Committed",
    "hash": "3B5467F0C33854C00AF89B04D5992474D58D770F7F364F262BEB593916823AD9"
  }
}
```

```
"header": {
      "timestamp": 1688285421141,
      "consensus_estimation": 4000,
      "height": 3,
      "view_change_index": 1,
      "previous_block_hash": "7EC52097164BD85DB87377EB9EBEF21B171A573F1BAADF91B2A251B45F3338BD",
      "transactions_hash": "90E6E72C095F1938408BDE7CF09BB2EEDE163E04B3AE813E875CC75F0F0D952B",
  ...
```

### Expected result

I expect  `transactions_hash` value should be identical to Pipeline event's transaction `hash` and the response of cli request.

### Logs in JSON format

<details>
  <summary>Log contents</summary>
  
  
</details>

**Comment by Erigara** [comment]
@rezacute how many transaction do you have in the block?

**Comment by rezacute** [comment]
@Erigara , i only have several transactions. I have tested from 0 transaction.

**Comment by mversic** [comment]
they are not the same because `./iroha_client_cli asset transfer` will return hash of `TransactionPayload` while pipeline event will return hash of `VersionedSignedTransaction`. The difference is that the later includes signatures when calculating hash while the former doesn't

**Comment by Erigara** [comment]
> @Erigara , i only have several transactions. I have tested from 0 transaction.

In case of more than one transaction would be expected because ` transactions_hash` is merkle tree root hash of the transactions in the block.

**Comment by rezacute** [comment]
@mversic `PipelineEvent` and `./iroha_client_cli asset transfer` is identical, but `./iroha_client_cli  blocks 1` is different

**Comment by rezacute** [comment]
@Erigara how can i obtain relation between `pipeline event` and `blocks` ?

**Comment by Erigara** [comment]
> @Erigara how can i obtain relation between `pipeline event` and `blocks` ?

Can you please clarify what kind of relation you want to obtain? Which block contain certain transaction?

**Comment by rezacute** [comment]
i expect `hash` result from  `./iroha_client_cli asset transfer` can be found from `block` stream

**Comment by Erigara** [comment]
> i expect `hash` result from `./iroha_client_cli asset transfer` can be found from `block` stream

Currently you can obtain hashes for transactions only by computing then, we will work on implementing this feature.

**Comment by rezacute** [comment]
> Currently you can obtain hashes for transactions only by computing

is it hard to compute? @Erigara

**Comment by mversic** [comment]
> is it hard to compute? @Erigara

it's the hash of the root of the merkle tree comprised of all transactions

**Comment by mversic** [comment]
closed in favor of #3670
